src/java/org/apache/cassandra/index/sai/disk/format/Version.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.format;
+
+import java.util.List;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Lists;
+
+import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.disk.v1.V1OnDiskFormat;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Format version of indexing component, denoted as [major][minor]. Same forward-compatibility rules apply as to
+ * {@link org.apache.cassandra.io.sstable.format.Version}.
+ */
+public class Version
+{
+    // Current version
+    public static final Version BA = new Version("ba", V1OnDiskFormat.instance, (c, i) -> defaultFileNameFormat(c, i, "ba"));
+
+    // These should be added in reverse order so that the latest version is used first. Version matching tests
+    // are more likely to match the latest version, so we want to test that one first.
+    public static final List<Version> ALL = Lists.newArrayList(BA);
```

Comment:
Would it make sense to make this a `SortedSet`, making the `Version` class implement `Comparable`? Something more or less like:
```java
public class Version implements Comparable<Version>
{
    public static final SortedSet<Version> ALL = new TreeSet<Version>(Comparator.reverseOrder()) {{
        ALL.add(BA);
    }};
...
    @Override
    public int compareTo(Version o)
    {
        return version.compareTo(o.version);
    }
```